pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -80,6 +83,7 @@
     protected volatile long incomingMessagesSize = 0;
     protected volatile Timeout batchReceiveTimeout = null;
     protected final Lock reentrantLock = new ReentrantLock();
+    private OrderedExecutor orderedExecutor = null;
```

Comment:
Can you just change the listenerExecutor to `OrderedExecutor`?

Reply:
Good suggestion!

Reply:
Oh @sijie, I have checked the details seems currently the listenerExecutor used by some other operations and the listener executor is a `ScheduledExecutorService`, the consumer will always get a listenerExecutor with 1 thread, so we can't change the `listenerExecutor` to `OrderedExecutor` directly.

Reply:
How about we use OrderedScheduler? BookKeeper common library also provides an `OrderedScheduler`.

Reply:
I have addressed the comment, please take a look.